test/Rings-conformance-tests.jl

```diff
@@ -147,50 +147,104 @@ function test_mutating_op_like_neg(f::Function, f!::Function, A)
 end
 
 function test_mutating_op_like_add(f::Function, f!::Function, A, B)
-   # initialize storage var with different values to check that its value is not used
-   for z in [zero(A), deepcopy(A), deepcopy(B)]
-      a = deepcopy(A)
-      b = deepcopy(B)
-      z = f!(z, a, b)
-      @test equality(z, f(A, B))
-      @test a == A
-      @test b == B
-   end
-
-   a = deepcopy(A)
-   b = deepcopy(B)
-   a = f!(a, a, b)
-   @test equality(a, f(A, B))
-   @test b == B
-
-   a = deepcopy(A)
-   b = deepcopy(B)
-   b = f!(b, a, b)
-   @test equality(b, f(A, B))
-   @test a == A
-
-   a = deepcopy(A)
-   b = deepcopy(B)
-   a = f!(a, b, b)
-   @test equality(a, f(B, B))
-   @test b == B
-
-   b = deepcopy(B)
-   b = f!(b, b, b)
-   @test equality(b, f(B, B))
+  # initialize storage var with different values to check that its value is not used
+  for z in [zero(A), deepcopy(A), deepcopy(B)]
+     a = deepcopy(A)
+     b = deepcopy(B)
+     z = f!(z, a, b)
+     @test equality(z, f(A, B))
+     @test a == A
+     @test b == B
+  end
+
+  a = deepcopy(A)
+  b = deepcopy(B)
+  a = f!(a, a, b)
+  @test equality(a, f(A, B))
+  @test b == B
+
+  a = deepcopy(A)
+  b = deepcopy(B)
+  b = f!(b, a, b)
+  @test equality(b, f(A, B))
+  @test a == A
+
+  a = deepcopy(A)
+  b = deepcopy(B)
+  a = f!(a, b, b)
+  @test equality(a, f(B, B))
+  @test b == B
+
+  b = deepcopy(B)
+  b = f!(b, b, b)
+  @test equality(b, f(B, B))
+
+  a = deepcopy(A)
+  b = deepcopy(B)
+  a = f!(a, b)
+  @test equality(a, f(A, B))
+  @test b == B
+
+  b = deepcopy(B)
+  b = f!(b, b)
+  @test equality(b, f(B, B))
+end
 
-   a = deepcopy(A)
-   b = deepcopy(B)
-   a = f!(a, b)
-   @test equality(a, f(A, B))
-   @test b == B
-
-   b = deepcopy(B)
-   b = f!(b, b)
-   @test equality(b, f(B, B))
+function test_mutating_op_like_mul(f::Function, f!::Function, A, B; left_factor_is_scalar::Bool = false, right_factor_is_scalar::Bool = false)
+  # initialize storage var with different values to check that its value is not used
+  for z in [zero(A), deepcopy(A), deepcopy(B)]
+     a = deepcopy(A)
+     b = deepcopy(B)
+     z = f!(z, a, b)
+     @test equality(z, f(A, B))
+     @test a == A
+     @test b == B
+  end
+
+  if !left_factor_is_scalar
+    a = deepcopy(A)
+    b = deepcopy(B)
+    a = f!(a, a, b)
+    @test equality(a, f(A, B))
+    @test b == B
+  end
+
+  if !right_factor_is_scalar
+    a = deepcopy(A)
+    b = deepcopy(B)
+    b = f!(b, a, b)
+    @test equality(b, f(A, B))
+    @test a == A
+  end
+
+  if !left_factor_is_scalar && !right_factor_is_scalar
+    a = deepcopy(A)
+    b = deepcopy(B)
+    a = f!(a, b, b)
+    @test equality(a, f(B, B))
+    @test b == B
+
+    b = deepcopy(B)
+    b = f!(b, b, b)
+    @test equality(b, f(B, B))
+  end
+
+  if !left_factor_is_scalar
+    a = deepcopy(A)
+    b = deepcopy(B)
+    a = f!(a, b)
+    @test equality(a, f(A, B))
+    @test b == B
+  end
+
+  if !left_factor_is_scalar && !right_factor_is_scalar
```

Comment:
```suggestion
  if A isa T && B isa T
```

Though actually... this doesn't involve `A` at all, so you could do
```suggestion
  if B isa T
```
or if you stick with the existing design:
```suggestion
  if !right_factor_is_scalar
```

Reply:
Maybe also merge this block with the one around line 227 testing `f!(b,b,b)` which is after all closely related to `f!(b,b)`.